### ISSUE TITLE
When --path is used, the directory's .swiftlint.yml is ignored

### ISSUE THREAD

**Issue by vincentisambart** [realm/SwiftLint#1631]
When using SwiftLint 0.20.0, when specifying a path (that is not the current directory) with `--path`, that directory's `.swiftlint.yml` is ignored.

To reproduce it, just create the following 2 files in a subdirectory.

```yml
# subdir/.swiftlint.yml
disabled_rules:
  - todo
```

```swift
// subdir/MyCode.swift
// TODO: ABCDEFG
```

Specifying the path causes the directory's `.swiftlint.yml` to be ignored.
```
$ swiftlint lint --no-cache --quiet # Not path specified, no warning as expected
$ swiftlint lint --no-cache --quiet --path subdir # Specifying the path gets you an unexpected warning
.../subdir/MyCode.swift:2:4: warning: Todo Violation: TODOs should be avoided (ABCDEFG). (todo)
```

**Comment by marcelofabri** [comment]
Thanks for the detailed issue. This is probably a side effect of #1604, in case someone wants to take a look at it.

I'll try to check it later.
